hmpps-kotlin-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/kotlin/auth/HmppsResourceServerConfiguration.kt

```diff
@@ -40,7 +40,9 @@ class HmppsResourceServerConfiguration {
       csrf { disable() }
       authorizeHttpRequests {
         customizer.unauthorizedRequestPathsCustomizer.unauthorizedRequestPaths.forEach { authorize(it, permitAll) }
-        authorize(anyRequest, authenticated)
+        customizer.anyRequestRoleCustomizer.defaultRole
+          ?.also { authorize(anyRequest, hasRole(it)) }
+          ?: also { authorize(anyRequest, authenticated) }
```

Comment:
I've seen this pattern in a few places - having a default role set for any auhtorised endpoints. Also it seems like a nice thing to have to make it easier to prevent endpoints without roles.

Reply:
I like and I don't like it. Not sure how you document it and how say Andy Lee's script would know all endpoints are protected - but lets keep this and see how it plays out